File: 백준 문제 풀이 모음/python/class 2/2775.py
import sys

test_case = int(sys.stdin.readline())

for _ in range(test_case) :
    k = int(sys.stdin.readline())
    n = int(sys.stdin.readline())
    
    ap = [[0] * n for _ in range(k+1) ]
    
    for i in range(k+1) :
        for j in range(n) :
            if i == 0 :
                ap[i][j] = j+1
            else :
                tmp = 0
                while True :
                    if tmp  == j + 1 :
                        break
                    ap[i][j] += ap[i-1][tmp]
                    tmp += 1
    
    print(ap[k][n-1])

File: 백준 문제 풀이 모음/kotlin/class 2/2775.kt

fun main() {
    val br = System.`in`.bufferedReader()
    val bw = System.`out`.bufferedWriter()

    val test_case = br.readLine().toInt()
    val sb = StringBuilder()
    repeat(test_case) {
        val k = br.readLine().toInt()
        val n = br.readLine().toInt()

        val arr = Array(k+1) {
            Array(n+1) { 0 }
        }

        for (i in 1..n) {
            arr[0][i] = i
//            println(arr[0][i])
        }


        for (i in 1..k) {
            for (j in 1..n) {
                for (x in 1..j) {
                    arr[i][j] += arr[i-1][x]
                }

            }
        }
        println(arr[k][n])
//        bw.write(arr[k][n].toString())

    }


}